src/IssueViz/Selection/AnalysisIssueNavigation.cs

```diff
@@ -74,14 +77,20 @@ private void NavigateToMatchingLocation(Func<IReadOnlyList<IAnalysisIssueLocatio
                 return;
             }
 
-            var navigateToLocation = order(currentFlow.Locations)
-                .FirstOrDefault(x =>
+            var navigableLocations = order(currentFlow.Locations)
```

Comment:
This is why I needed to replace the EventListener with a service that I can call: I want to keep checking until I find a navigable issue and update the selection service only at the end.